homeassistant/components/sunweg/config_flow.py

```diff
@@ -18,36 +20,49 @@ class SunWEGConfigFlow(ConfigFlow, domain=DOMAIN):
     def __init__(self) -> None:
         """Initialise sun weg server flow."""
         self.api: APIHelper = None
-        self.data: dict = {}
+        self.data: dict[str, Any] = {}
 
     @callback
-    def _async_show_user_form(self, errors=None) -> ConfigFlowResult:
+    def _async_show_user_form(self, step_id: str, errors=None) -> ConfigFlowResult:
         """Show the form to the user."""
+        default_username = ""
+        if CONF_USERNAME in self.data:
+            default_username = self.data[CONF_USERNAME]
         data_schema = vol.Schema(
             {
-                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_USERNAME, default=default_username): str,
                 vol.Required(CONF_PASSWORD): str,
             }
         )
 
         return self.async_show_form(
-            step_id="user", data_schema=data_schema, errors=errors
+            step_id=step_id, data_schema=data_schema, errors=errors
         )
 
+    def _set_auth_data(self, username: str, password: str) -> None:
+        """Set username and password."""
+        if self.api:
+            # Set username and password
+            self.api.username = username
+            self.api.password = password
+        else:
+            # Initialise the library with the username & password
+            self.api = APIHelper(username, password)
+
     async def async_step_user(self, user_input=None) -> ConfigFlowResult:
         """Handle the start of the config flow."""
         if not user_input:
-            return self._async_show_user_form()
-
-        # Initialise the library with the username & password
-        self.api = APIHelper(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
-        login_response = await self.hass.async_add_executor_job(self.api.authenticate)
-
-        if not login_response:
-            return self._async_show_user_form({"base": "invalid_auth"})
+            return self._async_show_user_form("user")
 
         # Store authentication info
         self.data = user_input
+        self._set_auth_data(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
+        try:
+            if not await self.hass.async_add_executor_job(self.api.authenticate):
+                return self._async_show_user_form("user", {"base": "invalid_auth"})
+        except SunWegApiError:
+            return self._async_show_user_form("user", {"base": "timeout_connect"})
```

Comment:
What does this change have to do with the reauth flow?

Reply:
It doesn't. I believe that is valid to threat API availability here and inside the reauth flow. Does it make any sense? I'll merge that inside the _set_auth_data method to avoid duped code. Does it sound better?